lib/node-pre-gyp.js

```diff
@@ -255,13 +254,7 @@ proto.setBinaryHostProperty = function(command) {
   if (command === 'publish') {
     target = 'staging_host';
   }
-  // each test is separate as opposed to combined like "if (this.rc[x] || this.opts[x])" to give
-  // priority to env var settings over command line.
-  if (this.rc['s3_host'] === 'staging') {
-    target = 'staging_host';
-  } else if (this.rc['s3_host'] === 'production') {
-    target = 'production_host';
-  } else if (this.opts['s3_host'] === 'staging') {
```

Comment:
related to previous point; i'll plan on a commit that just checks the env var in this section.

Reply:
👍  per email, please add as a followup. So I'll merge this as-is and let you fix up with `process.env` usage as you see fit (I don't have opinions on the naming)